src/cwe_checker_lib/src/checkers/cwe_252/context.rs

```diff
@@ -0,0 +1,330 @@
+//! Definition of the Taint Analysis for CWE252.
+//!
+//! Implementation of the [`TaintAnalysis`] for this CWE check. See the module
+//! documentation for more details on the algorithm and its limitations.
+
+use super::MustUseCall;
+
+use crate::analysis::fixpoint;
+use crate::analysis::forward_interprocedural_fixpoint::{
+    self, create_computation as fwd_fp_create_computation,
+};
+use crate::analysis::graph::{Graph as Cfg, HasCfg};
+use crate::analysis::interprocedural_fixpoint_generic::NodeValue;
+use crate::analysis::pointer_inference::{Data as PiData, PointerInference};
+use crate::analysis::taint::{state::State as TaState, TaintAnalysis};
+use crate::analysis::vsa_results::{HasVsaResult, VsaResult};
+use crate::intermediate_representation::{Blk, ExternSymbol, Jmp, Project, Term, Tid};
+use crate::utils::debug::ToJsonCompact;
+use crate::utils::log::CweWarning;
+
+use std::convert::AsRef;
+
+/// Type of the fixpoint computation of the taint analysis.
+pub type FpComputation<'a, 'b> = fixpoint::Computation<
+    forward_interprocedural_fixpoint::GeneralizedContext<'a, TaCompCtx<'a, 'b>>,
+>;
+
+impl ToJsonCompact for FpComputation<'_, '_> {
+    fn to_json_compact(&self) -> serde_json::Value {
+        let graph = self.get_graph();
+        let mut json_nodes = serde_json::Map::new();
+
+        for (node_index, node_value) in self.node_values().iter() {
+            let node = graph.node_weight(*node_index).unwrap();
+
+            if let NodeValue::Value(value) = node_value {
+                json_nodes.insert(format!("{node}"), value.to_json_compact());
+            }
+        }
+
+        serde_json::Value::Object(json_nodes)
+    }
+}
+
+/// Type that represents the definition of the taint analysis.
+///
+/// Values of this type represent the taint analysis for a particular call to an
+/// external function.
+pub struct TaCompCtx<'a, 'b: 'a> {
+    /// Extern function call that is analyzed.
+    call: MustUseCall<'a>,
+    project: &'a Project,
+    pi_result: &'a PointerInference<'b>,
+    /// Used to send generated CWE warnings to the collector.
+    cwe_sender: crossbeam_channel::Sender<CweWarning>,
+}
+
+impl<'a, 'b: 'a> TaCompCtx<'a, 'b> {
+    /// Creates a new taint analysis context for the given call to an external
+    /// function.
+    pub(super) fn new(
```

Comment:
But isn't the context module (and thus the context type) private to the `cwe_252` module anyway? In other words, `pub` and `pub(super)` should have the same effect. Which adds to my confusion, because explicitly using `pub(super)` here suggests otherwise.